dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamodbEnhancedClientRecorder.java

```diff
@@ -28,10 +47,49 @@ public BeanContainerListener setDynamoDbAsyncClient() {
             @Override
             public void created(BeanContainer container) {
                 DynamodbEnhancedClientProducer producer = container.beanInstance(DynamodbEnhancedClientProducer.class);
-                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class));
+                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class), createExtensionList());
             }
         };
 
         return beanContainerListener;
     }
+
+    private List<DynamoDbEnhancedClientExtension> createExtensionList() {
+
+        List<DynamoDbEnhancedClientExtension> extensions = new ArrayList<>();
+        for (String item : buildTimeConfig.clientExtensions.orElse(Collections.emptyList())) {
```

Comment:
As gsmet wrote earlier, they try to avoid using lambdas/streams here, cause of performance. Therefore I changed that to for-each. I think this is also easy to read and understand.